src/index.js

```diff
@@ -1,41 +1,194 @@
 'use strict'
 
 const Swarm = require('libp2p-swarm')
-const Peer = require('peer-info')
+const PeerId = require('peer-id')
+const PeerInfo = require('peer-info')
+const PeerBook = require('peer-book')
 const TCP = require('libp2p-tcp')
 // const UTP = require('libp2p-utp')
 const WS = require('libp2p-websockets')
 const spdy = require('libp2p-spdy')
 const multiaddr = require('multiaddr')
+const mafmt = require('mafmt')
 const EE = require('events').EventEmitter
 
 exports = module.exports
 
-exports.Node = function Node (peerInfo) {
+const OFFLINE_ERROR = new Error('The libp2p node is not started yet')
+
+exports.Node = function Node (pInfo, pBook) {
   if (!(this instanceof Node)) {
-    return new Node(peerInfo)
+    return new Node(pInfo, pBook)
   }
-  if (!peerInfo) {
-    peerInfo = new Peer()
-    peerInfo.multiaddr.add(multiaddr('/ip4/0.0.0.0/tcp/0'))
+
+  if (!pInfo) {
+    pInfo = new PeerInfo()
+    pInfo.multiaddr.add(multiaddr('/ip4/0.0.0.0/tcp/0'))
   }
 
+  if (!pBook) {
+    pBook = new PeerBook()
+  }
+
+  this.peerInfo = pInfo
+  this.peerBook = pBook
+
   // Swarm
-  this.swarm = new Swarm(peerInfo)
-  this.swarm.transport.add('ws', new WS())
-  this.swarm.transport.add('tcp', new TCP())
+  this.swarm = new Swarm(pInfo)
   this.swarm.connection.addStreamMuxer(spdy)
   this.swarm.connection.reuse()
 
-  this.start = (done) => {
-    this.swarm.listen(done)
+  this.swarm.on('peer-mux-established', (peerInfo) => {
+    this.peerBook.put(peerInfo)
+  })
+
+  this.swarm.on('peer-mux-closed', (peerInfo) => {
+    this.peerBook.removeByB58String(peerInfo.id.toB58String())
+  })
+
+  this.isOnline = false
```

Comment:
I would hide this to make sure it's not accidentally changed from the outside. Easiest to do with a getter and a locally scoped variable.